tests/integration/relations/opensearch_provider/helpers.py

```diff
@@ -2,13 +2,55 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+import asyncio
+import logging
+from typing import Optional
+
+import yaml
 from pytest_operator.plugin import OpsTest
 from tenacity import RetryError, Retrying, stop_after_delay, wait_fixed
 
 
-async def check_new_relation(ops_test: OpsTest):
-    """Smoke test to check relation is online."""
-    raise NotImplementedError
+async def get_application_relation_data(
+    ops_test: OpsTest,
+    application_name: str,
+    relation_name: str,
+    key: str,
+    relation_id: str = None,
+) -> Optional[str]:
+    """Get relation data for an application.
+
+    Args:
+        ops_test: The ops test framework instance
+        application_name: The name of the application
+        relation_name: name of the relation to get connection data from
+        key: key of data to be retrieved
+        relation_id: id of the relation to get connection data from
+
+    Returns:
+        the data that was requested or None
+            if no data in the relation
+
+    Raises:
+        ValueError if it's not possible to get application unit data
+            or if there is no data for the particular relation endpoint
+            and/or alias.
+    """
+    unit_name = f"{application_name}/0"
```

Comment:
nit: Do you think it would make sense to pass the unit id in the arguments of the function? so that you're not constrained by unit 0

Reply:
Agreed, I'm now passing the `unit_id` instead